File: packages/types/src/compra.ts
import type { ISODateString, Monto, UUID } from './common';

export interface DetalleCompra {
  id: UUID;
  compraId: UUID;
  productoId: UUID;
  cantidad: number;
  precioUnitario: Monto;
}

export interface Compra {
  id: UUID;
  clienteId: UUID;
  usuarioId: UUID;
  subtotal: Monto;
  descuento: Monto;
  total: Monto;
  fecha: ISODateString;
}

export interface CompraConDetalle extends Compra {
  detalles: DetalleCompra[];
}

export interface Canje {
  id: UUID;
  clienteId: UUID;
  beneficioId: UUID;
  condicionId: UUID;
  compraId: UUID;
  usuarioId: UUID;
  descuentoAplicado: Monto;
  fecha: ISODateString;
}

File: apps/api/internal/domain/compra.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Compra es la cabecera de una venta registrada (historial del cliente).
type Compra struct {
	ID        uuid.UUID
	ClienteID uuid.UUID
	UsuarioID uuid.UUID
	Subtotal  int
	Descuento int
	Total     int
	Fecha     time.Time
}
